cassandraprotocol/frame/decoder.go

```diff
@@ -117,6 +130,21 @@ func (c *Codec) DiscardBody(bodyLength int32, source io.Reader) (err error) {
 	return err
 }
 
+// ReadBody reads the contents of a frame body without decoding them. It is illegal to call this method before calling
+// DecodeHeader.
+func (c *Codec) ReadBody(bodyLength int32, source io.Reader) (body []byte, err error) {
```

Comment:
Isn't it a bit better to return `*bytes.Buffer` ?

Reply:
Imo `bytes.Buffer` is for cases where it's expected that the buffer will be modified which is not the case here.

Reply:
> Imo `bytes.Buffer` is for cases where it's expected that the buffer will be modified which is not the case here.

+1

Reply:
Hmm isn't it the opposite? The caller can modify `[]byte` and mess with its contents. With `bytes.Buffer` that's less possible (but still possible). If modification is a concern, we could return `io.Reader` then. What seems odd to me is that this library never uses `[]byte` anywhere (except internally in compressors), this makes me think that `[]byte` is too specific.

Reply:
It's possible with both `bytes.Buffer` and `[]byte` but that's not the issue, it's fine if the caller mess with its contents, this module doesn't have any ownership over that data after it is read anyway. What I'm saying is that the expected scenario is that the caller doesn't mess with its contents (I don't see a reason to) so returning a `bytes.Buffer` might send the wrong message that somehow the caller is supposed to modify it.

About `io.Reader`, I don't have a strong opinion against it but it doesn't feel right to return a `Reader` interface on a method that is supposed to perform the read operation. I'll add another comment about this here https://github.com/datastax/go-cassandra-native-protocol/pull/5#discussion_r513280843

Reply:
If we change this to `io.Reader` doesn't that mean that there will be 2 allocations for the same `[]byte` regardless? The `io.Reader` will have its underlying array and when the caller wants to read the actual body, it will create another `[]byte`.

Reply:
> What seems odd to me is that this library never uses []byte anywhere (except internally in compressors), this makes me think that []byte is too specific.

This is the only place where the library returns a "raw" portion of the frame, everywhere else the API returns and receives fully decoded frames and messages.